pkg/fieldpath/fieldpath.go

```diff
@@ -18,26 +18,40 @@ package fieldpath
 
 import (
 	"fmt"
+	"sort"
+	"strconv"
 	"strings"
+	"unsafe"
 
 	"k8s.io/apimachinery/pkg/api/meta"
-	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
 // FormatMap formats map[string]string to a string.
 func FormatMap(m map[string]string) (fmtStr string) {
 	// output with keys in sorted order to provide stable output
-	keys := sets.NewString()
-	for key := range m {
-		keys.Insert(key)
+	keys := make([]string, 0, len(m))
+	var grow int
+	for k, v := range m {
+		keys = append(keys, k)
+		// why + 2? (=,\n)
+		// this is just a vague size to avoid frequent requests for space
+		// why is it 3*len(v)/2, from strconv.quoteWith
+		grow += len(k) + len(v) + 3*len(v)/2 + 2
 	}
-	for _, key := range keys.List() {
-		fmtStr += fmt.Sprintf("%v=%q\n", key, m[key])
+	sort.Strings(keys)
+	// allocate space to avoid expansion
+	dst := make([]byte, 0, grow)
+	for _, key := range keys {
+		if len(dst) > 0 {
+			dst = append(dst, '\n')
+		}
+		dst = append(dst, key...)
+		dst = append(dst, '=')
+		dst = strconv.AppendQuote(dst, m[key])
 	}
-	fmtStr = strings.TrimSuffix(fmtStr, "\n")
-
-	return
+	// form strings.Builder{}.String()
+	return *(*string)(unsafe.Pointer(&dst))
```

Comment:
Thanks for the PR, but I don't think we should switch this to use `unsafe`. From what I can see, this function is not used in a hotpath that justifies the added complexity / safety issues.

Reply:
Yes indeed, it does not take into account this situation